packages/apps/plugins/plugin-navtree/src/NavTreePlugin.tsx

```diff
@@ -43,14 +44,23 @@ export type NavTreePluginProvides = SurfaceProvides &
   GraphBuilderProvides &
   TranslationsProvides;
 
+type NavTreeState = { root?: NavTreeItemGraphNode };
+
+type NavTreeSettings = { openItemIds?: OpenItemIds };
+
 export const NavTreePlugin = (): PluginDefinition<NavTreePluginProvides> => {
-  const state = create<{ root?: NavTreeItemGraphNode; openItemIds?: OpenItemIds }>({});
+  const state = create<NavTreeState>({});
+
+  const settings = new LocalStorageStore<NavTreeSettings>('dxos.org/settings/navtree', {
+    openItemIds: { root: true, 'dxos.org/plugin/space-spaces': true, 'dxos.org/plugin/files': true },
+  });
 
   let graphPlugin: Plugin<GraphProvides> | undefined;
   let graph: Graph | undefined;
 
   const handleOpenItemIdsChange = (nextOpenItemIds: OpenItemIds) => {
-    state.openItemIds = nextOpenItemIds;
+    // TODO(thure): This might become a localstorage leak; openItemIds that no longer exist should be removed from this map.
```

Comment:
if they're wholly overwritten every time they change then it's a bounded leakage and doesn't seem like a problem?

Reply:
It’s not wholly overwritten, I don’t think.

Reply:
> settings.values.openItemIds = nextOpenItemIds

does this not replace the whole array? how does the leakage happen? does `nextOpenItemIds` leak forward item ids that aren't in the tree if they were passed in via props?

Reply:
> does nextOpenItemIds leak forward item ids that aren't in the tree if they were passed in via props?

Yes.